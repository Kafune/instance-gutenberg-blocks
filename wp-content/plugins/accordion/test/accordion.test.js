

import Accordion from "./../src/components/accordion";
import React from "react";
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


describe("Accordion text editing", () => {
	const testAccordion = {
		heading: "Title that can be expanded",
		description:
			"By creating an Account on our service, you agree to subscribe to newsletters, marketing or promotional materials and other information we may send. However, you may opt out of receiving any, or all, of these communications from us by following the unsubscribe link or instructions provided in any email we send.",
	};

	beforeEach(() => {});

	test("Return default text value", () => {
		const wrapper = shallow(
			<Accordion
				heading={testAccordion.heading}
				description={testAccordion.description}
				changeProp={(property) => changeProp({ property }, i)}
			/>
		);

		// const accordion = jest.mock('./../src/components/accordion',() => 'Accordion')

		// expect(wrapper.find(Accordion).prop('heading')).toBe("Title that can be expanded");
		// const container = wrapper.find(Accordion).length
		// console.log(toJson(wrapper).find(Accordion).length)
		expect(toJson(wrapper)).toMatchSnapshot();

	});
});
