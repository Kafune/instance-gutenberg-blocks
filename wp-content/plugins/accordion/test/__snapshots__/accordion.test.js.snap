// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion text editing Return default text value 1`] = `
<div
  class="accordion"
>
  <div
    class="accordion-heading"
  >
    <ForwardRef(RichTextWrapper)
      onChange={[Function]}
      placeholder="Title..."
      tagName="h4"
      value="Title that can be expanded"
    />
    <span
      class="accordion-icon"
    >
      -
    </span>
  </div>
  <div
    class="accordion-description accordion-open"
  >
    <ForwardRef(RichTextWrapper)
      onChange={[Function]}
      placeholder="Enter your description..."
      tagName="p"
      value="By creating an Account on our service, you agree to subscribe to newsletters, marketing or promotional materials and other information we may send. However, you may opt out of receiving any, or all, of these communications from us by following the unsubscribe link or instructions provided in any email we send."
    />
  </div>
</div>
`;
